libraries/AP_RangeFinder/AP_RangeFinder_LightWareSerial.cpp

```diff
@@ -44,7 +48,7 @@ bool AP_RangeFinder_LightWareSerial::get_reading(float &reading_m)
             if (c == '\r') {
                 linebuf[linebuf_len] = 0;
                 const float dist = strtof(linebuf, nullptr);
-                if (!is_negative(dist)) {
+                if (!is_negative(dist) && dist * 100.0f < distance_cm_max + LIGHTWARE_LOST_SIGNAL_ADD_CM) {
```

Comment:
I think this will not work actually because it will mean that if the vehicle climbs above the RNGFNDx_MAX_CM value the lidar's status will become NoData when really we would want it to be "OutOfRangeHigh".

.. so I think we need a change that rejects specific values (<0, 130m, 250m).  We should probably add a new method called something like "lost_signal_dist(float dist)" that returns true if the distance is one of these special values.

Reply:
@shiv-tyagi did have a branch to do the above that him and I looked at together. But the crux of the issue was determining the model info. of the rangefinder: by either reading the info by requesting it from the rangefinder and parsing and creating new model parameters for those radars which we cant request the info from.  The other issue was getting someone who has the specified radars test the changes.

Reply:
@hendjoshsr71, txs.  I think we shouldn't bother with the complexity of trying to determine the model.  It would be nice if we could do that but I think it will change too often and will end up being unreliable.  Better just to catch these very specific values regardless of model.

Reply:
@rmackay9 I probably am misunderstanding, but the method get_reading(float &reading_m) from the driver still returns true even if all the measurements were deemed 'invalid', and reading_m gets set to out-of-range-high value:
`
    // all readings were invalid so return out-of-range-high value
    if (invalid_count > 0) {
        reading_m = MIN(MAX(LIGHTWARE_DIST_MAX_CM, max_distance_cm() + LIGHTWARE_OUT_OF_RANGE_ADD_CM), UINT16_MAX) * 0.01f;
        return true;
    }
`
Doesn't this mean that when the vehicle climbs above the RNGFNDx_MAX_CM value the lidar's status will correctly become "OutOfRangeHigh"?

Reply:
Ah, right.  So I guess the impact would be that we wouldn't report or log the distances that are beyond RNGFNDx_MAX_CM?  I think at the moment we do but the rangefinder's status changes (to OutOfRangeHigh).  I still tend to think that a direct check of the specific values that mean lost-signal is the best approach.